gnovm/cmd/gno/test.go

```diff
@@ -500,10 +512,25 @@ func runTestFiles(
 	n := gno.MustParseFile("main_test.gno", testmain)
 	m.RunFiles(n)
 
+	printedEvents := 0
+
 	for _, test := range testFuncs.Tests {
 		testFuncStr := fmt.Sprintf("%q", test.Name)
 
-		eval := m.Eval(gno.Call("runtest", testFuncStr))
+		res := gno.Call("runtest", testFuncStr)
+
+		eval := m.Eval(res) // NOTE: verbose prints get here
+		if printEvents {
+			ctx := m.Context.(stdlibs.ExecContext)
+
+			events := ctx.EventLogger.Events()
+			for _, ev := range events[printedEvents:] {
```

Comment:
Are the events accumulated between different `gno.Call`s and `m.Eval`s, and that's why you're skipping previously printed events?

Reply:
Yes it is.